components/03-organisms/galleries/media-grid/yds-media-grid-interactive.js

```diff
@@ -0,0 +1,42 @@
+Drupal.behaviors.mediaGridInteractive = {
+  attach(context) {
+    const mediaGrids = context.querySelectorAll('.media-grid');
+
+    /**
+     * toggleModalState
+     * @function toggleModalState toggles modal state
+     */
+    const toggleModalState = (state) => {
+      return state === 'inactive' ? 'active' : 'inactive';
+    };
+
+    mediaGrids.forEach((grid) => {
+      const modalState = grid.getAttribute('data-media-grid-modal-state');
+      const items = grid.querySelectorAll('.media-grid__item');
+
+      items.forEach((item) => {
+        item.addEventListener('click', () => {
+          grid.setAttribute(
+            'data-media-grid-modal-state',
+            toggleModalState(modalState),
+          );
+        });
+      });
+    });
+
```

Comment:
Consider adding a keyboard trap to this file. I noticed I'm immediately able to tab-outside of the modal - https://github.com/emulsify-ds/compound/pull/35/files#diff-266cf28551632cfda8819f7b4b2b2a449d4fe849742c74188265dab8205c81bfR26

Reply:
Fo sho... all of this interactive stuff is actually coming in a follow-up PR (already heavily underway) I think I just accidentally committed some of the initial work to this branch instead of that one... Oops!

I'll add the keyboard trap in the next PR.